roscala/src/main/scala/coop/rchain/roscala/prim/Prim.scala

```diff
@@ -5,13 +5,20 @@ import coop.rchain.roscala.Vm.State
 import coop.rchain.roscala.ob._
 import coop.rchain.roscala.prim.fixnum.fxPlus
 import coop.rchain.roscala.prim.rblfloat.flPlus
+import coop.rchain.roscala.util.misc.{numberSuffix, properPrep}
+
+import scala.reflect.{classTag, ClassTag}
 
 abstract class Prim extends Ob {
   def fn(ctxt: Ctxt, globalEnv: GlobalEnv): Ob
 
-  // TODO: Add error case
-  def dispatchHelper(state: State, globalEnv: GlobalEnv): Ob =
-    fn(state.ctxt, globalEnv)
+  def dispatchHelper(state: State, globalEnv: GlobalEnv): Ob = {
+    val n = state.ctxt.nargs
+    if (minArgs <= n && n <= maxArgs)
```

Comment:
I'm doing the checking of number of argument here like rosette do. Should I use macro as before?

Reply:
Seems like the macro is not really required.